qubespdfconverter/server.py

```diff
@@ -185,15 +208,122 @@ def __init__(self, path):
         self.path = path
         self.pagenums = 0
         self.batch = None
+        self.password = ""
+
+    def _read_password(self, password_success):
+        if not password_success:
+            cmd = ["zenity", "--title", "File protected by password", "--password"]
+            self.password = subprocess.run(cmd, capture_output=True, check=True)\
+                .stdout.split(b"\n")[0]
+
+
+    def _decrypt(self):
+        """
+        Try to remove the password of a libreoffice-compatible file,
+        and store the resulting file in INITIAL_NAME.nopassword.
+        The steps are:
+        - Connect to a libreoffice API server, listening on localhost on port 2202
+        - Try to load a document with additionnal properties:
+              - "Hidden" to not load any libreoffice GUI
+              - "Password" to automatically try to decrypt the document
+        - Store the document without additionnal properties [this remove the password]
+        """
+
+        src = "file://"+str(self.path)
+        dst = "file://"+str(self.path)+".nopassword"
+
+        local_context = uno.getComponentContext()
+        resolver = local_context.ServiceManager.createInstanceWithContext(
+            "com.sun.star.bridge.UnoUrlResolver",
+            local_context
+        )
+        ctx = resolver.resolve(
+            "uno:socket,host=localhost,port=2202;urp;StarOffice.ComponentContext"
+        )
+        smgr = ctx.ServiceManager
+        desktop = smgr.createInstanceWithContext("com.sun.star.frame.Desktop", ctx)
+
+        hidden_property = PropertyValue()
+        hidden_property.Name = "Hidden"
+        hidden_property.Value = True
+
+        password_property = PropertyValue()
+        password_property.Name = "Password"
+        password_property.Value = self.password
+
+        document = desktop.loadComponentFromURL(
+            src,
+            "_blank",
+            0,
+            (password_property, hidden_property,)
+        )
+        document.storeAsURL(dst, ())
 
 
     async def sanitize(self):
         """Start sanitization tasks"""
-        self.pagenums = self._pagenums()
-        self.batch = asyncio.Queue(self.pagenums)
 
-        send(self.pagenums)
+        password_success = False
```

Comment:
I don't really like flags like this. I assume the `zenity` command in `_read_password()` returns something  on failure, even if it's just an empty string? Maybe we can work around that instead of using this flag.